crates/wast/src/ast/wast.rs

```diff
@@ -263,3 +301,20 @@ impl<'a> Parse<'a> for QuoteModule<'a> {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::parser::ParseBuffer;
+    use crate::ast::wast::{WastDirective, WastCheckType};
+
+    #[test]
+    fn assert_canonical_nan() {
+        let buffer = ParseBuffer::new("assert_return_canonical_nan (invoke \"foo\" (f32.const 0)) f32x4").unwrap();
+        let parser = buffer.parser();
```

Comment:
I think this could use the standalone `parse` function perhaps instead of using `buffer.parser()`?

Reply:
Ah, I see it now; makes sense.